cmds/fusermount/fusermount_test.go

```diff
@@ -0,0 +1,155 @@
+	}
+	skip := len("2018/12/20 16:54:31 ")
+
+	for _, v := range tab {
+		c := testutil.Command(t, v.a...)
+		c.Env = append(c.Env, v.env...)
+		c.Stdin = bytes.NewReader([]byte(v.n))
+		o, err := c.CombinedOutput()
+		// log.Fatal exits differently on circleci and real life
+		// Even an explicit os.Exit(1) returns with a 2. WTF?
+		if err := testutil.IsExitCode(err, v.e); err != nil {
+			t.Error(err)
+			continue
+		}
+		if len(o) < skip {
+			t.Errorf("Fusermount %v %v: want '%v', got '%v'", v.n, v.a, v.o, o)
+			continue
+		}
+		out := string(o[len("2018/12/20 16:54:31 "):])
+		if out != v.o {
+			t.Errorf("Fusermount %v %v: want '%v', got '%v'", v.n, v.a, v.o, out)
+			continue
+		}
+	}
+}
+
+func TestMount(t *testing.T) {
+	if os.Getuid() != 0 {
+		t.Skip("Skipping, not root")
+	}
+	// Get a socketpair to talk on, then spawn the kid
+	fds, err := unix.Socketpair(syscall.AF_FILE, syscall.SOCK_STREAM, 0)
+	if err != nil {
+		t.Fatal(err)
+	}
+	t.Logf("fds are %v", fds)
+
+	writeFile := os.NewFile(uintptr(fds[0]), "fusermount-child-writes")
+	defer writeFile.Close()
+
+	readFile := os.NewFile(uintptr(fds[1]), "fusermount-parent-reads")
+	defer readFile.Close()
+
+	tmpDir, err := ioutil.TempDir("", "TestFusermount")
+	if err != nil {
+		t.Fatal("TempDir failed: ", err)
+	}
+	defer os.RemoveAll(tmpDir)
+
+	fc, err := net.FileConn(readFile)
+	if err != nil {
+		t.Fatalf("FileConn from fusermount socket: %v", err)
+	}
+	defer fc.Close()
+
+	uc, ok := fc.(*net.UnixConn)
+	if !ok {
+		t.Fatalf("unexpected FileConn type; expected UnixConn, got %T", fc)
+	}
+
+	c := testutil.Command(t, "-v", tmpDir)
+	c.Env = append(c.Env, "_FUSE_COMMFD=3")
+	c.ExtraFiles = []*os.File{writeFile}
+	c.SysProcAttr = &syscall.SysProcAttr{Cloneflags: syscall.CLONE_NEWNS}
+	go func() {
+		o, err := c.CombinedOutput()
+		t.Logf("Running fuse: %v,%v", string(o), err)
+	}()
+
+	buf := make([]byte, 32) // expect 1 byte
+	oob := make([]byte, 32) // expect 24 bytes
+	_, oobn, _, _, err := uc.ReadMsgUnix(buf, oob)
+	if err != nil {
+		t.Fatalf("uc.ReadMsgUnix: got %v, want nil", err)
+	}
+	t.Logf("ReadMsgUnix returns oobn %v, err %v", oobn, err)
+	scms, err := syscall.ParseSocketControlMessage(oob[:oobn])
+	if err != nil {
+		t.Fatalf("syscall.ParseSocketControlMessage(%v): got %v, want nil", oob[:oobn], err)
+	}
+	t.Logf("syscall.ParseSocketControlMessage(%v): returns %v", oob[:oobn], scms)
+	if len(scms) != 1 {
+		t.Fatalf("expected 1 SocketControlMessage; got scms = %#v", scms)
+	}
+	scm := scms[0]
+	gotFds, err := syscall.ParseUnixRights(&scm)
+	if err != nil {
+		t.Fatalf("syscall.ParseUnixRights: %v", err)
+	}
+	if len(gotFds) != 1 {
+		t.Fatalf("wanted 1 fd; got %#v", gotFds)
+	}
+	f := os.NewFile(uintptr(gotFds[0]), "/dev/fuse")
+	t.Logf("file to fuse is %v", f)
+	// Now every good program should unmount.
+	c = testutil.Command(t, "-v", "-u", tmpDir)
+	o, err := c.CombinedOutput()
+	t.Logf("Running fuse: %v,%v", string(o), err)
+	if err != nil {
+		t.Fatal(err)
+	}
+}
```

Comment:
Add another newline after function.

Reply:
so, my feeling here is that if gofmt does not do it, then it should be acceptable. But I'll add it.